Standards/scs-0100-v4-flavor-naming.md

```diff
@@ -0,0 +1,521 @@
+
+It may also be important to make the CPU generation clearly recognizable, as this is always a topic in
+discussions with customers.
+
+Note that not all relevant properties of flavors can be discovered; creating a specification
+to address this is a separate but related effort to the name standardization.
+Commonly used infrastructure-as-code tools do not provide a way to use discoverability
+features to express something like "I want a flavor with 2 vCPUs, 8GiB of RAM, a local
+20GB SSD disk and Infiniband support but I don't care whether it's AMD or intel" in a
+reasonable manner. Using flavor names to express this will thus continue to be useful
+and we don't expect the need for standardization of flavor names to go away until
+the commonly used IaC tools work on a higher abstraction layer than they currently do.
+
+## Design Considerations
+
+### Type of information included
+
+From discussions of our operators with their customers we learned that
+the following characteristics are important in a flavor description:
+
+| Type              | Description                                            |
+| :---------------- | :----------------------------------------------------- |
+| Generation        | CPU Generation                                         |
+| Number of CPU     | Number of vCPUs - suffixed by L,V,T,C (see below)      |
+| Amount of RAM     | Amount of memory available for the VM                  |
+| Performance Class | Ability to label high-performance CPUs, disks, network |
+| CPU Type          | X86-intel, X86-amd, ARM, RISC-V, Generic               |
+| "bms"             | Bare Metal System (no virtualization/hypervisor)       |
+
+This list is likely not comprehensive and will grow over time.
+
+Rather than using random names `s5a.medium` and assigning a discrete set of properties
+to them, we wanted to come up with a scheme that allows to systematically derive
+names from properties and vice versa. The scheme allows for short names (by not
+encoding all details) as well as very detailed longer names.
+
+## Complete Proposal for systematic flavor naming
+
+| Prefix | CPUs & Suffix     | RAM[GiB]           | optional: Disk[GB]&type       | opt: extensions |
+| ------ | ----------------- | ------------------ | ----------------------------- | ----------------|
+| `SCS-` | N`L/V/T/C`\[`i`\] | `-`N\[`u`\]\[`o`\] | \[`-`\[M`x`\]N\[`n/s/l/p`\]\] | \[`_`EXT\]      |
+
+Note that N and M are placeholders for numbers here.
+The optional fields are denoted in brackets (and have opt: in the header.
+See below for extensions.
+
+Note that all letters are case-sensitive.
+
+Typical flavor names look like `SCS-4V-16-50` for a flavor with 4vCPUs (with limited
+oversubscription), 16GiB RAM and a 50GB disk (of unspecified type).
+
+## Proposal Details
+
+### [REQUIRED] CPU Suffixes
+
+Next to the number of vCPUs, these vCPUs need to be characterized ti describe their nature.
```

Comment:
I will incorporate typo corrections such as this one. Other discussions should be postponed, because this PR is **only meant for splitting** the document. (Be aware that all  your remarks apply to the original document.)

Reply:
--> <https://github.com/SovereignCloudStack/standards/issues/327>